.github/workflows/publish-docs.yml

```
@@ -74,6 +62,7 @@ jobs:
         run: mkdocs build
 
       - uses: actions/upload-artifact@da838ae9595ac94171fa2d4de5a2f117b3e7ac32 # renovate: tag=v2.3.0
+        if: github.event_name == 'push' && github.ref == 'refs/heads/main'
         with:
           name: site
           path: site
```

```suggestion
          path: site
          if-no-files-found: error
```

The default behavior for `actions/upload-artifact` is to `warn` on `if-no-files-found` [^docs], which means that we could have zero files, and it would just happily proceed to the `Publish Docs` step.

I think it makes sense to abort the upload of the artifact, and error out, if there are no files present, as that means something has gone very badly wrong.

[^docs]: https://github.com/actions/upload-artifact#customization-if-no-files-are-found